odoo/addons/base/models/ir_fields.py

```diff
@@ -203,8 +210,80 @@ def _str_to_json(self, model, field, value):
             raise self._format_import_error(ValueError, msg, value)
 
     def _str_to_properties(self, model, field, value):
-        msg = _("Unable to import field type '%s'  ", field.type)
-        raise self._format_import_error(ValueError, msg)
+        if isinstance(value, str):  # Get be already convert into list by _extract_records
+            try:
+                value = json.loads(value)
+            except ValueError:
+                msg = _("'%(value)s' does not seem to be a valid JSON for field '%%(field)s'")
+                raise self._format_import_error(ValueError, msg, {'value': value})
+
+        if not isinstance(value, list):
+            msg = _("'%(value)s' does not seem to be a valid Properties value for field '%%(field)s'. A Properties should contain a list of property. Each property need at least 'name', 'type' and 'string' attribute")
+            raise self._format_import_error(ValueError, msg, {'value': value})
+
+        warnings = []
+        for property_value in value:
+            if {'name', 'type', 'string'} > property_value.keys():
```

Comment:
Sets are partially-ordered, you cannot just inverse the operator.
```suggestion
            if not ({'name', 'type', 'string'} <= property_value.keys()):
```

Reply:
The order of set isn't take in account for operators right ? Or I don't understand your comments ?

Reply:
The order operators like `>` for sets mean "is a strict subset". So this is false when you have `property_value = {'a'}` which is not what you want: you must have all 3 attributes.

If you find it more readable, you can express your condition as `not all(f in property_value for f in ('name', 'type', 'string'))`.

Reply:
> So this is false when you have property_value = {'a'} which is not what you want

Ho yeah, I wrote that too fast. Indeed!